frontend/pages/hosts/ManageHostsPage/components/HostsFilterBlock/HostsFilterBlock.tsx

```diff
@@ -0,0 +1,445 @@
+
+import PencilIcon from "../../../../../../assets/images/icon-pencil-14x14@2x.png";
+import TrashIcon from "../../../../../../assets/images/icon-trash-14x14@2x.png";
+import PolicyIcon from "../../../../../../assets/images/icon-policy-fleet-black-12x12@2x.png";
+
+const baseClass = "hosts-filter-block";
+
+// TODO: need to look improving the props API for this component
+interface IHostsFilterBlockProps {
+  munkiIssueDetails: IMunkiIssuesAggregate | null;
+  policyResponse: PolicyResponse;
+  selectedLabel?: ILabel;
+  policyId?: any;
+  policy?: IPolicy;
+  macSettingsStatus?: any;
+  softwareId?: any;
+  mdmId?: number;
+  mdmEnrollmentStatus?: any;
+  lowDiskSpaceHosts?: number;
+  osId?: any;
+  osName?: any;
+  osVersion?: any;
+  munkiIssueId?: number;
+  // TODO: request for this data can happen here or lower in component tree in os filter block
+  osVersions?: IOperatingSystemVersion[];
+  isOnlyObserver?: boolean;
+  softwareDetails: ISoftware | null;
+  mdmSolutionDetails: IMdmSolution | null;
+  // TODO: change to onHandleClearRouteParam
+  // QUESTION: include router as param?
+  handleClearRouteParam: () => void;
+  handleClearFilter: (omitParams: string[]) => void;
+  onChangePoliciesFilter: (response: PolicyResponse) => void;
+  onChangeMacSettingsFilter: (
+    newMacSettingsStatus: MacSettingsStatusQueryParam
+  ) => void;
+  onClickEditLabel: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+  onClickDeleteLabel: () => void;
+}
+
+/**
+ * Renders the filtering section of the Manage Hosts Page. This will handle rendering
+ * the correct filter pills and any filter dropdowns associated with those pills.
+ */
+const HostsFilterBlock = ({
+  selectedLabel,
+  policyId,
+  macSettingsStatus,
+  softwareId,
+  mdmId,
+  mdmEnrollmentStatus,
+  lowDiskSpaceHosts,
+  osId,
+  osName,
+  osVersion,
+  munkiIssueId,
+  munkiIssueDetails,
+  policyResponse,
+  osVersions,
+  isOnlyObserver,
+  softwareDetails,
+  policy,
+  mdmSolutionDetails,
+  handleClearRouteParam,
+  handleClearFilter,
+  onChangePoliciesFilter,
+  onChangeMacSettingsFilter,
+  onClickEditLabel,
+  onClickDeleteLabel,
+}: IHostsFilterBlockProps) => {
+  const handleClearPoliciesFilter = () => {
+    handleClearFilter(["policy_id", "policy_response"]);
+  };
+
+  const handleClearOSFilter = () => {
+    handleClearFilter(["os_id", "os_name", "os_version"]);
+  };
+
+  const handleClearMacSettingsStatusFilter = () => {
+    handleClearFilter(["macos_settings"]);
+  };
+
+  const handleClearSoftwareFilter = () => {
+    handleClearFilter(["software_id"]);
+  };
+
+  const handleClearMDMSolutionFilter = () => {
+    handleClearFilter(["mdm_id"]);
+  };
+
+  const handleClearMDMEnrollmentFilter = () => {
+    handleClearFilter(["mdm_enrollment_status"]);
+  };
+
+  const handleClearMunkiIssueFilter = () => {
+    handleClearFilter(["munki_issue_id"]);
+  };
+
+  const handleClearLowDiskSpaceFilter = () => {
+    handleClearFilter(["low_disk_space"]);
+  };
+
```

Comment:
What if we refactor the `FilterPill` interface:

```
interface IFilterPillProps {
  label: string;
  icon?: string;
  tooltipDescription?: string | ReactNode;
  className?: string;
  paramNames: string[];
  handleClearParams: (paramNames: string[]) => void;
}
...
const FilterPill = ({
  label,
  icon,
  tooltipDescription,
  className,
  paramNames,
  handleClearParams,
}: IFilterPillProps) => {
...
const onClear = () => handleClearParams(paramNames);
...

```

Then you could remove all the handlers above and this component could become more declarative:

```
...
const { pillLabel, description, paramNames, paramType } = HOSTS_FILTER_CONFIG[activeFilter];
const onClear = paramType === "route" ? handleClearRouteParam ? handleClearFilter;

return (
          <FilterPill
            label={label}
            tooltipDescription={description}
            paramNames={paramNames}
            onClear={onClear}
          />
          ...

```

Reply:
yes good idea. I'll make this change.

Reply:
Looking at the new implementation, Im hesitant to give `FilterPill` more responsibility other than displaying the pill and calling an `onClear` handler. I think keeping it a purely presentational component makes more sense if you look at the usage of `FilterPill` [here](https://github.com/ghernandez345/fleet/blob/b7adf77210ab6d834eb11580d39d37264830f21a/frontend/pages/hosts/ManageHostsPage/components/HostsFilterBlock/HostsFilterBlock.tsx#L139-L143
).

The `onClear` handler has slightly different behaviour than all the other filter pills and wouldn't use the `paramNames` argument, and it calls `handleClearRouteParams` instead of `handleClearFilter`. I wouldn't want to move the calling of the correct handler into `FilterPill` and feel the component doesn't need internal knowledge about the query params it manages.

How do you feel about reducing the code by defining the handlers inline instead of as variables?

```tsx
<FilterPill
  label={label}
  tooltipDescription={TooltipDescription}
  onClear={() => handleClearFilter(["os_id", "os_name", "os_version"])}
/>
```